src/Microsoft.AspNet.Authentication/DataHandler/TicketSerializer.cs

```diff
@@ -36,98 +39,177 @@ public virtual AuthenticationTicket Deserialize(byte[] data)
+            {
+                WriteClaim(writer, claim);
+            }
+
+            var bootstrap = identity.BootstrapContext as string;
+            if (!string.IsNullOrEmpty(bootstrap))
+            {
+                writer.Write(true);
+                writer.Write(bootstrap);
+            }
             else
             {
-                writer.Write(principal.Identities.Count());
-                foreach (var identity in principal.Identities)
-                {
-                    var authenticationType = string.IsNullOrEmpty(identity.AuthenticationType) ? string.Empty : identity.AuthenticationType;
-                    writer.Write(authenticationType);
-                    WriteWithDefault(writer, identity.NameClaimType, DefaultValues.NameClaimType);
-                    WriteWithDefault(writer, identity.RoleClaimType, DefaultValues.RoleClaimType);
-                    writer.Write(identity.Claims.Count());
-                    foreach (var claim in identity.Claims)
-                    {
-                        WriteWithDefault(writer, claim.Type, identity.NameClaimType);
-                        writer.Write(claim.Value);
-                        WriteWithDefault(writer, claim.ValueType, DefaultValues.StringValueType);
-                        WriteWithDefault(writer, claim.Issuer, DefaultValues.LocalAuthority);
-                        WriteWithDefault(writer, claim.OriginalIssuer, claim.Issuer);
-                    }
-
-                    var bootstrap = identity.BootstrapContext as string;
-                    if (string.IsNullOrEmpty(bootstrap))
-                    {
-                        writer.Write(0);
-                    }
-                    else
-                    {
-                        writer.Write(bootstrap.Length);
-                        writer.Write(bootstrap);
-                    }
-                }
+                writer.Write(false);
+            }
+
+            if (identity.Actor != null)
+            {
+                writer.Write(true);
+                WriteIdentity(writer, identity.Actor);
+            }
+            else
+            {
+                writer.Write(false);
             }
-            PropertiesSerializer.Write(writer, model.Properties);
         }
 
-        public static AuthenticationTicket Read([NotNull] BinaryReader reader)
+        public virtual void WriteClaim([NotNull] BinaryWriter writer, [NotNull] Claim claim)
+        {
+            WriteWithDefault(writer, claim.Type, claim.Subject?.NameClaimType ?? ClaimsIdentity.DefaultNameClaimType);
+            writer.Write(claim.Value);
+            WriteWithDefault(writer, claim.ValueType, ClaimValueTypes.String);
+            WriteWithDefault(writer, claim.Issuer, ClaimsIdentity.DefaultIssuer);
+            WriteWithDefault(writer, claim.OriginalIssuer, claim.Issuer);
+
+            // Write the number of properties contained in the claim.
+            writer.Write(claim.Properties.Count);
```

Comment:
`Claim.Properties` is not supposed to be null, but old versions of <Mono 4.0 were known to not initialize this property. I chatted with @davidfowl and he agreed it was not worth it.